src/RockBridge.cpp

```diff
@@ -29,172 +25,159 @@
-{
-	std::cout << " ~ " << std::endl;
-	gzmsg << "RockBridge: initializing model: "<< (model)->GetName() << std::endl;
+    void RockBridge::createTask(gazebo::physics::WorldPtr world, gazebo::physics::ModelPtr model)
+    {
+	    std::cout << " ~ " << std::endl;
+	    gzmsg << "RockBridge: initializing model: "<< (model)->GetName() << std::endl;
 
-	// Create and initialize one rock component for each gazebo model
-    ModelTask* task = new gazebo::ModelTask();
-    task->setGazeboModel(world, model, environment);
-    setupTaskActivity(task);
-}
+	    // Create and initialize one rock component for each gazebo model
+        gazebo::ModelTask* task = new gazebo::ModelTask();
+        task->setGazeboModel(world, model);
+        setupTaskActivity(task);
+    }
 //======================================================================================
-// Callback method triggered every update begin
-// It test conditions and implement all rock components functionalities
-void RockBridge::updateBegin(common::UpdateInfo const& info)
-{
-    for(Activities::iterator it = activities.begin(); it != activities.end(); ++it)
+//  Callback method triggered every update begin
+//  It test conditions and implement all rock components functionalities
+//======================================================================================
+    void RockBridge::updateBegin(gazebo::common::UpdateInfo const& info)
     {
-        (*it)->trigger();
+        for(Activities::iterator it = activities.begin(); it != activities.end(); ++it)
+        {
+            (*it)->trigger();
+        }
     }
-}
-
 //======================================================================================
-void RockBridge::updateEnd()
-{
-}
+    void RockBridge::updateEnd()
+    {
+    }
 //======================================================================================
-RockBridge::RockBridge()
-{
-}
+    RockBridge::RockBridge()
+    {
+    }
 //======================================================================================
-RockBridge::~RockBridge()
-{
-	// Delete pointers to activity
-	for(Activities::iterator activity_it = activities.begin(); 
-		    activity_it != activities.end(); ++activity_it)
-	{
-		delete *activity_it;
-	}
-	activities.clear();
-	
-	// Delete pointers to tasks
-	for(Tasks::iterator task_it = tasks.begin();
-		    task_it != tasks.end(); ++task_it)
-	{
-		delete *task_it;
-	}
-	tasks.clear();
+    RockBridge::~RockBridge()
+    {
+	    // Delete pointers to activity
+	    for(Activities::iterator activity_it = activities.begin(); 
+		        activity_it != activities.end(); ++activity_it)
+	    {
+		    delete *activity_it;
+	    }
+	    activities.clear();
 	
-	worlds.clear();
+//	    // Delete pointers to tasks
+//	    for(Tasks::iterator task_it = tasks.begin();
+//		        task_it != tasks.end(); ++task_it)
+//	    {
+//		    delete *task_it;
+//	    }
+//	    tasks.clear();
+//	
+//	    worlds.clear();
```

Comment:
You must delete the pointers. What you don't have to do is clear the vectors, as they get destroyed with the object.